test/spec/ui/overlay.spec.js

```diff
@@ -0,0 +1,29 @@
+(function () {
+    'use strict';
+
+    define(
+        [
+            'chai',
+            'ui/overlay'
+        ],
+        function (chai, overlay) {
+            var expect = chai.expect;
+
+            describe('The `overlay` module', function () {
+                it('Defines a function', function () {
+                    expect(overlay).to.be.a('function');
+                });
+                describe('When invoked', function () {
+                    var instance;
+                    beforeEach(function () {
+                        instance = overlay();
+                    });
+                    it('Returns an object with `loading` and `error` methods', function () {
+                        expect(instance.loading).to.be.a('function');
```

Comment:
This only tests that the functions exist, not what those functions do.  However in this specific case, what the functions do is some DOM manipulation, and without mocking out the DOM, it's not possible to test this.